pkg/apis/meta/v1/types.go

```diff
@@ -91,6 +91,14 @@ type OwnerReference struct {
 	// If true, this reference points to the managing controller.
 	// +optional
 	Controller *bool `json:"controller,omitempty" protobuf:"varint,6,opt,name=controller"`
+	// If true, AND if the owner has the "DeletingDependents" finalizer, then
+	// the owner cannot be deleted from the key-value store until this
+	// reference is removed.
+	// Defaults to false.
+	// To set this field, a user needs "delete" permission of the owner,
+	// otherwise 422 (Unprocessable Entity) will be returned.
+	// +optional
+	BlockOwnerDeletion *bool `json:"blockOwnerDeletion,omitempty" protobuf:"varint,7,opt,name=blockOwnerDeletion"`
```

Comment:
Requiring both the owner and child resource to agree on synchronous collection makes sense to me, so I'm ok with this.

As you pointed out in person, in the common case of controller-created resources, this would always be true, so that the choice of synchronous vs async. deletion would be controlled upon deletion of the owner.

I assume that if this were set to true, any additions to ownerReferences would need to do the permissions check, and could fail with permission denied.

I agree with @lavalamp that "block" is an overloaded term, but I can't think of a better one, so I'm fine with this field name.